src/pages/change-password-page/change-password-page.ts

```diff
@@ -0,0 +1,142 @@
+import './change-password-page.css';
+import Handlebars from 'handlebars';
+import { v4 } from 'uuid';
+import { changePasswordPageTemplate } from './change-password-page.tmpl';
+import { Routes } from '../../constants/routes';
+import { Input } from '../../components/input/input';
+import { Button } from '../../components/button/button';
+import { Link } from '../../components/link/link';
+import { IComponentProps } from '../../lib/interfaces/component-props.interface';
+import { Block } from '../../lib/Block/Block';
+import {
+  maxLength,
+  minLength,
+  password,
+  validationMessage,
+} from '../../presentation-logic/forms/validate-input';
+import { getFormData } from '../../presentation-logic/forms/get-form-data';
+
+interface IChildren {
+  appInputOldPassword: Input;
+  appInputPassword: Input;
+  appInputPasswordRepeat: Input;
+  appButtonSave: Button;
+  appLinkToAccountPage: Link;
+}
+
+interface IChangePasswordPageProps extends IComponentProps {
+  children?: IChildren;
+}
+
+const accountPageLink = `/${Routes.ACCOUNT}`;
+const formId = `i${v4()}`;
+const formSelector = `#${formId}`;
+
+export class ChangePasswordPage extends Block<IChangePasswordPageProps> {
+  private _oldPassword: string = '';
+
+  private _password: string = '';
+
+  private _passwordRepeat: string = '';
+
+  private _childrenComponents: IChildren = {
+    appInputOldPassword: new Input({
+      name: 'old-password',
+      label: 'Старый пароль',
+      type: 'password',
+      internalEvents: {
+        input: {
+          blur: () => this._handleOldPasswordChange(),
```

Comment:
Когда нет надобности прокидывать доп. аргументы, то незачем создавать анонимные обертки и такие записи можно сократить
```suggestion
          blur: this._handleOldPasswordChange,
```

Reply:
Джейк Арчибальт, кажется, советует так не делать
https://jakearchibald.com/2021/function-callback-risks/

Reply:
Хорошее замечание. Стоит учитывать. Вообще там имеется ввиду код, который не под нашим контролем
> unless they're designed for it
Когда мы используем либу, мы не знаем под что они заточены или будут... Здесь другое. Обработчики которые мы делаем, предназначены быть колбэками в ответ на клики и тд. Зато такие функции можно отписать, без доп. телодвижений в виде AbortController. Так же можно написать. 
```
function foo(e) { ... }
button.onclick = foo
```
То есть коллбэк наш и меняем его мы - design for being callback так сказать

Reply:
Постоянно контролировать, используется здесь внешняя зависимость, обернутая нашим сервисом, или это "исконно" наш пакет, кажется мне порочной практикой. Лучше всегда писать единообразно.